### ISSUE TITLE
Missing pointer nullification when removing a json value.

### ISSUE THREAD

**Issue by tsbg** [jacobwilliams/json-fortran#477]
Hi

I was trying to move a `json_value` from one structure to another one by first removing it with `json%remove()`  without destroying the value and then adding it to the other structure with `json%add_by_path()`. I got some surprising results, for example missing values.

After looking into the `json_value_remove()` procedure, I am pretty sure that the parent, previous and next pointers in the `json_value` p must be nullified. Currently only the next pointer is nullified. I think this leads to unexpected behaviour for example in `json%destory()` or `json%add_by_path()`.

Can you confirm my observations or do I misunderstand something?

**Comment by jacobwilliams** [comment]
I can look into this. If you could post an example case where the behavior is unexpected that would help a lot. I think I have a unit test for this, but maybe I didn't account for some cases?

**Comment by jacobwilliams** [comment]
Looking at it.. I think you are correct. I'll build some more tests cases and look into it further. Thanks for reporting this!

**Comment by tsbg** [comment]
Sorry for the delay. Here is a small example that should show the problem:

```Fortran
program test
  !
  use :: json_module, CK => json_CK
  !
  implicit none
  !
  character(kind=CK,len=*), parameter :: STR = CK_'&
      &{&
      &  "object1": {&
      &    "value-1-10": 1,&
      &    "value-1-20": 2,&
      &    "value-1-30": 3,&
      &    "value-1-40": 4,&
      &    "value-1-50": 5 &
      &  },&
      &  "object2": {&
      &    "value-2-10": 10,&
      &    "value-2-20": 11,&
      &    "value-2-30": 12,&
      &    "value-2-40": 13,&
      &    "value-2-50": 14 &
      &  }&
      &}'
  !
  character(len=:), allocatable :: errtxt
  type(json_core)               :: json
  type(json_value), pointer     :: inp, p, p2
  logical                       :: found, status_ok
  !
  call json%deserialize(inp, STR)
  call json%check_for_errors(status_ok, errtxt)
  if (.not. status_ok) then
    write (*, *) errtxt
    stop
  end if
  !
  ! 1. This works as expected.
  call json%get(inp, 'object1.value-1-30', p, found)
  if (found) then
    call json%clone(p, p2)
    call json%remove(p, .true.)
    call json%add_by_path(inp, 'object2.value-2-35', p2)
  end if
  call json%print(inp)
  !
  ! Result:
  !  {
  !    "object1": {
  !      "value-1-10": 1,
  !      "value-1-20": 2,
  !      "value-1-40": 4,
  !      "value-1-50": 5
  !    },
  !    "object2": {
  !      "value-2-10": 10,
  !      "value-2-20": 11,
  !      "value-2-30": 12,
  !      "value-2-40": 13,
  !      "value-2-50": 14,
  !      "value-2-35": 3
  !    }
  !  }  
  !
  print *, '-------'
  !
  ! 2. This one fails.
  call json%get(inp, 'object1.value-1-40', p, found)
  if (found) then
    call json%remove(p, .false.)
    call json%add_by_path(inp, 'object2.value-2-45', p)
  end if
  call json%print(inp)
  !
  ! Result:
  !  {
  !    "object1": {
  !      "value-1-10": 1,
  !      "value-1-20": 2          // "value-1-50": 5 is missing here!
  !    },
  !    "object2": {
  !      "value-2-10": 10,
  !      "value-2-20": 11,
  !      "value-2-30": 12,
  !      "value-2-40": 13,
  !      "value-2-50": 14,
  !      "value-2-35": 3,
  !      "value-2-45": 4
  !    }
  !  }
  !
end program test
```

**Comment by jacobwilliams** [comment]
I think it's fixed now. Will merge as soon as I get a chance.

**Comment by tsbg** [comment]
Thank you very much for fixing this, I really appreciate it!

**Comment by jacobwilliams** [comment]
Thanks for letting me know. This fix is now in `8.2.1`.
